addon/webextension/catcher.js

```diff
@@ -42,27 +44,29 @@ window.catcher = (function () {
   /** Wrap the function, and if it raises any exceptions then call unhandled() */
   exports.watchFunction = function watchFunction(func) {
     return function () {
-      var result;
       try {
-        result = func.apply(this, arguments);
+        return func.apply(this, arguments);
       } catch (e) {
         exports.unhandled(e);
         throw e;
       }
-      return result;
     };
   };
 
   exports.watchPromise = function watchPromise(promise) {
     return promise.catch((e) => {
-      console.error("------Error in promise:", e+"");
+      console.error("------Error in promise:", e);
       console.error(e.stack);
       exports.unhandled(makeError(e));
       throw e;
     });
   };
 
   exports.registerHandler = function (h) {
+    if (handler) {
+      console.error("registerHandler called after handler was already registered");
```

Comment:
We should have `catcher.unhandled(new Exception("registerHandler called after handler was already registered"))`